nixops/resources/__init__.py

```diff
@@ -70,14 +70,15 @@ def show_type(self) -> str:
         return self.get_type()
 
 
-ResourceDefinitionType = TypeVar(
-    "ResourceDefinitionType", bound="ResourceDefinition", contravariant=True
-)
+ResourceDefinitionType = TypeVar("ResourceDefinitionType", bound="ResourceDefinition")
```

Comment:
I think this is nice, actually, since we really did want it to be invariant I think.